shared/src/balancer/event_handler.rs

```diff
@@ -151,11 +166,16 @@ impl PoolRegistry {
             .collect()
     }
 
-    async fn insert_events(&mut self, events: Vec<(EventIndex, PoolCreated)>) -> Result<()> {
+    async fn insert_events(
+        &mut self,
+        events: Vec<(EventIndex, PoolCreated)>,
```

Comment:
One way to reduce duplicated code would be to define a common type that both contract events turn into. Here a simple `(u64, H160)` would suffice.

Reply:
I like the `PoolCreated` struct - I must have missed it.